tasks/agent.py

```diff
@@ -129,6 +129,12 @@ def build(
         python_runtimes=python_runtimes,
     )
 
+    if sys.platform == 'darwin' and platform.processor() == 'arm64':
+        # Using amd64 when building with Mac M1
+        env["GOARCH"] = "amd64"
+        # Important for x-compiling
```

Comment:
So that future people have an easier time than us, let's link to the issue where I saw this was needed

```suggestion
        # Important for x-compiling; see https://github.com/golang/go/issues/39398#issuecomment-638898309
```